### ISSUE TITLE
Guard's Attacking Groups.

### ISSUE THREAD

**Issue by Taiphoz** [CitizensDev/Citizens#560]
Hay all.

Hope this has not been resolved I had a look and noted other people having similar issues so think it should still be needing work.

I have Permissionsbukkit, and of course citizens

I am trying to make a Pirate, so I tried making a guard, and turning on his aggro, setting him to flag the admin,member and default groups, in a nut shell any player on the server, but he will not attack.

I even tried /guard setflag -p player and it would still not attack the other player.

If some one attacked the npc he would then run after them, but I need them to aggro from the start.

I have build a big pirate ship, and am trying to build a raid type deal, full of pirates that will attack the players, and if they manage to hack their way through there will be a chest at the end, or a boss to kill.

at least thats what I wana get working .. but with them not auto attacking other players its kinda not working.

here is my profile for the pirate.

'6':
  guard:
    flags:
      creeper:
        priority: '4'
        safe: 'false'
        type: MOB
      member:
        priority: '3'
        type: GROUP
        safe: 'false'
      admin:
        priority: '2'
        type: GROUP
        safe: 'false'
      default:
        priority: '1'
        type: GROUP
        safe: 'false'
    type: 'GUARD'
    radius: '10.0'
    aggressive: 'true'
    toggle: 'true'
  basic:
    location: kilda2,-238.40625,121.0,-495.09375,-2.602562,0.0
    owner: Scourge6
    color: '15'
    look-when-close: 'true'
    items: 268:0,298:0,299:0,300:0,301:0,
    inventory: 268/1/0,AIR*35,
    talk-when-close: 'false'
    balance: '0.0'
    talk: 'true'
    name: Pirate_Pete

group names are correct, same case all lower.

thanks for any help. hope this gets resolve I have big plans if this plugin starts to really shine. it will change the face of minecraft.

**Comment by Thass** [comment]
Did you try to make the guard a bouncer? Do this with /guard bouncer.
I had the same problems as you and making him a bouncer solved it. =)
Bouncers are more areaprotecting guards so this should be what you seek.

**Comment by Taiphoz** [comment]
/npc create killer
/toggle guard
/guard aggro
/bouncer
/guard addflag 1 -g all

That should make the guard attack all players on the server yes? I did also try addflag 1 -p playername but that didnt work either.

**Comment by ghost** [comment]
/guard addflag 1 -a

that should do the trick. Are your player's able to hurt the guards at all, because mine cannot.

**Comment by Taiphoz** [comment]
yeah if they hit the guard it will defend itselfs, but it wont just go attack a player thats walking past.

**Comment by Taiphoz** [comment]
I got them attacking players, which is cool cos now I can make my pirate ship raid type thing, but they wont attack mobs and players at the same time. so it seems like its one or the other.

**Comment by Cixxar** [comment]
PLEASE comment on how you done this ! we are having problems with citizens too. they wont attack or more

**Comment by Taiphoz** [comment]
I found that if you create a new citizen from scratch with the sole purpose of attacking players it will work.

/npc create guy
/toggle guard
/aggro
/guard addflag -a

I might have missed something off the top of my head but that should do it, for us it makes our guards attack all players, appart from the npc owner, but it WILL NOT attack mobs.

Even when we add mobs to its flags it still will only attack players.

So seems that they will either go for players, or mobs and not both, if you setup your npc to attack mobs first you will need to delete it and start over if you want it attacking players.

at least thats what works for us.
